test/signalcorr_complex.jl

```diff
@@ -0,0 +1,153 @@
+@test autocov(x1) ≈ acovx1
+@test autocov(cmplx_x1) ≈ acovx1
+@test autocov(x)  ≈ [autocov(x1) autocov(x2)]
+@test autocov(cmplx_x)  ≈ [autocov(cmplx_x1) autocov(cmplx_x2)]
+
+acorx1 = [1.0 + 0.0im,
+          -0.007061066965510023 - 0.24670570770539213im,
+          -0.3791703080554228 + 0.18499330626611243im,
+          -0.07251256107537019 + 0.17200449686941222im,
+          -0.006618732813301651 + 0.031065686027770673im,
+          0.1569606471499575 - 0.0038094765432061303im,
+          -0.00678395250709688 - 0.106474871528861im,
+          -0.18656146869859214 + 0.04664475073114351im,
+          0.05282671316002114 + 0.0065117700502952056im,
+          -0.051079270194684986 - 0.0538075492419246im]
+
+@test autocor(x1) ≈ acorx1
+@test autocor(cmplx_x1) ≈ acorx1
+@test autocor(x)  ≈ [autocor(x1) autocor(x2)]
+@test autocor(cmplx_x)  ≈ [autocor(cmplx_x1) autocor(cmplx_x2)]
+
+
+# crosscov & crosscor
+
+rcov0 = [0.96 + 0.32im,
+         -0.96 - 0.32im,
+         0.24 + 0.08im,
+         -1.44 - 0.48im,
+         0.0 + 0.0im,
+         1.44 + 0.48im,
+         -0.24 - 0.08im,
+         0.96 + 0.32im,
+         -0.96 - 0.32im]
+
+@test crosscov([1+9im, 2+7im, 3+5im, 4+3im, 5+1im], [1-im, -1+im, 1-im, -1+im, 1-im]) ≈ rcov0
+@test crosscov([1:5 ;], [1:5;]) ≈ [-0.8, -0.8, -0.2, 0.8, 2.0, 0.8, -0.2, -0.8, -0.8]
+
+c11 = crosscov(x1, x1)
+c12 = crosscov(x1, x2)
+c21 = crosscov(x2, x1)
+c22 = crosscov(x2, x2)
+@test crosscov(cmplx_x1, cmplx_x2) ≈ c12
+
+@test crosscov(x,  x1) ≈ [c11 c21]
+@test crosscov(cmplx_x, cmplx_x1) ≈ [c11 c21]
+@test crosscov(x1, x)  ≈ [c11 c12]
+@test crosscov(cmplx_x1, cmplx_x)  ≈ [c11 c12]
+@test crosscov(x,  x)  ≈ cat([c11 c21], [c12 c22], dims=3)
+@test crosscov(cmplx_x,  cmplx_x)  ≈ cat([c11 c21], [c12 c22], dims=3)
+
+rcor0 = [0.230940107675850,
+        -0.230940107675850,
+         0.057735026918963,
+        -0.346410161513775,
+         0.000000000000000,
+         0.346410161513775,
+        -0.057735026918963,
+         0.230940107675850,
+        -0.230940107675850]
+
+@test crosscor([1, 2, 3, 4, 5], [1, -1, 1, -1, 1]) ≈ rcor0
+@test crosscor([1:5;], [1:5;]) ≈ [-0.4, -0.4, -0.1, 0.4, 1.0, 0.4, -0.1, -0.4, -0.4]
+
+c11 = crosscor(x1, x1)
+c12 = crosscor(x1, x2)
+c21 = crosscor(x2, x1)
+c22 = crosscor(x2, x2)
+@test crosscor(cmplx_x1, cmplx_x2) ≈ c12
+
+@test crosscor(x,  x1) ≈ [c11 c21]
+@test crosscor(cmplx_x, cmplx_x1) ≈ [c11 c21]
+@test crosscor(x1, x)  ≈ [c11 c12]
+@test crosscor(cmplx_x1, cmplx_x)  ≈ [c11 c12]
+@test crosscor(x,  x)  ≈ cat([c11 c21], [c12 c22], dims=3)
+@test crosscor(cmplx_x, cmplx_x)  ≈ cat([c11 c21], [c12 c22], dims=3)
+
+
+## pacf least squares
+pacf_ls = [-1.598495044296996e-03 - 2.915104118351207e-01im
+           -5.560162016912027e-01 + 2.950837739894279e-01im
+           -2.547001916363494e-02 + 2.326084658014266e-01im
+           -5.427443903358727e-01 + 3.146715147305132e-01im]
+
+@test pacf(x1, 1:4) ≈ pacf_ls[1:4]
+
+## pacf Yule-Walker
+
+function yulewalker_qr(v::AbstractVector)
```

Comment:
Can you add a comment explaining what these do? Why don't we check this for reals too?